www/js/metrics/metricsHelper.ts

```diff
@@ -29,24 +32,34 @@ export const secondsToMinutes = (seconds: number) => formatForDisplay(seconds /
 export const secondsToHours = (seconds: number) => formatForDisplay(seconds / 3600);
 
 // segments metricsDays into weeks, with the most recent week first
-export function segmentDaysByWeeks(days: DayOfMetricData[], nWeeks?: number) {
-  const weeks: DayOfMetricData[][] = [];
-  for (let i = days?.length - 1; i >= 0; i -= 7) {
-    weeks.push(days.slice(Math.max(i - 6, 0), i + 1));
-  }
-  if (nWeeks) return weeks.slice(0, nWeeks);
-  return weeks;
+export function segmentDaysByWeeks(days: DayOfMetricData[], lastDate: string) {
+  const weeks: DayOfMetricData[][] = [[]];
+  let cutoff = isoDateWithOffset(lastDate, -7 * weeks.length);
+  [...days].reverse().forEach((d) => {
+    const date = dateForDayOfMetricData(d);
+    // if date is older than cutoff, start a new week
+    if (isoDatesDifference(date, cutoff) > 0) {
+      weeks.push([]);
+      cutoff = isoDateWithOffset(lastDate, -7 * weeks.length);
+    }
+    weeks[weeks.length - 1].push(d);
+  });
```

Comment:
Could we use a regular `for` loop instead of `forEach`? This way, we wouldn't need to reverse the array. Please let me know if this change would have any other effects!

```js
  for(let i=days.length-1; i>=0; i--) {
    const date = dateForDayOfMetricData(days[i]);
    // if date is older than cutoff, start a new week
    if (isoDatesDifference(date, cutoff) > 0) {
      weeks.push([]);
      cutoff = isoDateWithOffset(lastDate, -7 * weeks.length);
    }
    weeks[weeks.length - 1].push(days[i]);
  }
```

Reply:
Great suggestion! I think this will have the same effect and be a bit more performant